services/src/main/java/org/exoplatform/addons/gamification/utils/Utils.java

```diff
@@ -201,18 +208,25 @@ public static UserInfo getUserById(Long id, Long challengeId) {
   }
 
   public static UserInfo createUser(Identity identity, Space space, List<Long> managersId) {
-    SpaceService spaceService = CommonsUtils.getService(SpaceService.class);
     UserInfo userInfo = new UserInfo();
     userInfo.setAvatarUrl(identity.getProfile().getAvatarUrl());
     userInfo.setFullName(identity.getProfile().getFullName());
     userInfo.setRemoteId(identity.getRemoteId());
     userInfo.setId(identity.getId());
+    String userId = getCurrentUser();
     if (space != null) {
-      userInfo.setManager(spaceService.isManager(space, getCurrentUser()));
-      userInfo.setMember(spaceService.isMember(space, getCurrentUser()));
-      userInfo.setRedactor(spaceService.isRedactor(space, getCurrentUser()));
-      userInfo.setCanAnnounce(canAnnounce(space.getId()));
-      userInfo.setCanEdit(canEditChallenge(managersId, space.getId()));
+      SpaceService spaceService = CommonsUtils.getService(SpaceService.class);
+      Boolean isSuperManager =  spaceService.isSuperManager(userId);
+      boolean isManager = isSuperManager || spaceService.isManager(space, userId);
+      boolean isMember = spaceService.isMember(space, userId);
+      boolean isRedactor = spaceService.isRedactor(space, userId) ;
+      boolean hasRedactor = spaceService.isRedactor(space,userId) ;
+      Boolean isChallengeOwner = managersId.stream().anyMatch(i -> i == Long.parseLong(identity.getId()));
+      userInfo.setManager(isManager);
+      userInfo.setMember(isMember);
+      userInfo.setRedactor(isRedactor);
+      userInfo.setCanAnnounce(hasRedactor ? isRedactor || isManager : isMember);
+      userInfo.setCanEdit(isChallengeOwner && isManager);
```

Comment:
User can not edit his own challenge ?

Reply:
yes for those who are no more space manager  can't edit his challenge